lib/auth/join_iam.go

```diff
@@ -68,21 +75,33 @@ const (
 // To keep this validation simple and secure, we check the given endpoint
 // against a static list of known valid endpoints. We will need to update this
 // list as AWS adds new regions.
-func validateSTSHost(stsHost string) error {
+func validateSTSHost(stsHost string, cfg *iamRegisterConfig) error {
 	valid := apiutils.SliceContainsStr(validSTSEndpoints, stsHost)
-	if valid {
-		return nil
+	if !valid {
+		return trace.AccessDenied("IAM join request uses unknown STS host %q. "+
+			"This could mean that the Teleport Node attempting to join the cluster is "+
+			"running in a new AWS region which is unknown to this Teleport auth server. "+
+			"Alternatively, if this URL looks suspicious, an attacker may be attempting to "+
+			"join your Teleport cluster. "+
+			"Following is the list of valid STS endpoints known to this auth server. "+
+			"If a legitimate STS endpoint is not included, please file an issue at "+
+			"https://github.com/gravitational/teleport. %v",
+			stsHost, validSTSEndpoints)
+	}
+
+	if cfg.fips && !apiutils.SliceContainsStr(fipsSTSEndpoints, stsHost) {
+		if cfg.authVersion.LessThan(semver.Version{Major: 12}) {
+			log.Warnf("Non-FIPS STS endpoint (%s) was used by a node joining "+
+				"the cluster with the IAM join method. "+
+				"Ensure that all nodes joining the cluster are up to date and also run in FIPS mode. "+
+				"This will be an error in Teleport 12.0.0.",
+				stsHost)
```

Comment:
Kinda unfortunate that this will only be shown in the logs but there doesn't seem to be any place closer to the actual user interaction where we could show this. As far as I understand the [Joining Nodes via AWS IAM Role](https://goteleport.com/docs/management/guides/joining-nodes-aws-iam/) guide, this code path will be triggered only when a Teleport agent tries to join the cluster, won't it?

Reply:
That is correct. I'm not sure there's any better place to warn about it, but in theory nothing should break as long as everything is within 1 major version.